web/pdf_cursor_tools.js

```diff
@@ -0,0 +1,182 @@
+ * @property {EventBus} eventBus - The application event bus.
+ */
+
+/**
+ * @class
+ */
+var PDFCursorTools = (function PDFCursorToolsClosure() {
+  /**
+   * @constructs PDFCursorTools
+   * @param {PDFCursorToolsOptions} options
+   */
+  function PDFCursorTools(options) {
+    var self = this;
+    this.container = options.container;
+    this.eventBus = options.eventBus;
+
+    this.active = CursorTool.SELECT;
+    this.activeBeforePresentationMode = null;
+
+    this.handTool = new GrabToPan({
+      element: this.container,
+    });
+
+    this._addEventListeners();
+
+    // Attempt to convert the old 'enableHandToolOnLoad' preference to the new
+    // 'cursorToolOnLoad' preference. Note that this is only done when the new
+    // preference has not yet been set to a non-default value.
+    // TODO: Remove this fallback after a suitable time has passed.
+    Promise.all([Preferences.get('enableHandToolOnLoad'),
+                 Preferences.get('cursorToolOnLoad')]).then(function (values) {
+      var handToolPref = values[0];
+      var cursorToolPref = values[1];
+
+      if (handToolPref === true) {
+        // Reset the old preference.
+        Preferences.set('enableHandToolOnLoad', false);
+        // Update the new preference.
+        if (cursorToolPref === CursorTool.SELECT) {
+          cursorToolPref = CursorTool.HAND;
+          Preferences.set('cursorToolOnLoad', cursorToolPref);
+        }
+      }
+      self.switchTool(cursorToolPref);
+    }).catch(function (reason) { });
+  }
+
+  PDFCursorTools.prototype = {
+    /**
+     * @returns {number} One of the values in `CursorTool`.
+     */
+    get activeTool() {
+      return this.active;
+    },
+
+    /**
+     * NOTE: This method is ignored while Presentation Mode is active.
+     * @param {number} tool - The cursor mode that should be switched to,
+     *                        must be one of the values in `CursorTool`.
+     */
+    switchTool: function PDFCursorTools_switchTool(tool) {
+      if (this.activeBeforePresentationMode !== null) {
+        return; // Cursor tools cannot be used in Presentation Mode.
+      }
+      if (tool === this.active) {
+        return; // The requested tool is already active.
+      }
+
+      // De-activate the active tool.
+      switch (this.active) {
+        case CursorTool.HAND:
+          this.handTool.deactivate();
+          break;
+      }
+
+      // Activate the new tool.
+      switch (tool) {
+        case CursorTool.SELECT:
+          break;
+        case CursorTool.HAND:
+          this.handTool.activate();
+          break;
+        default:
+          console.error('PDFCursorTools_switchTools: "' + tool +
+                        '" is an unsupported value.');
+          return;
+      }
+      // Update the active tool *after* it has been validated above,
+      // in order to prevent setting it to an invalid state.
+      this.active = tool | 0;
```

Comment:
Do we need `| 0` here? As far as I can see the switch above enforces that `tool` is either `CursorTool.SELECT` or `CursorTool.HAND`, which are both integers already.